src/rust/storage/seg/src/hashtable/mod.rs

```diff
@@ -722,3 +888,45 @@ impl HashTable {
         hasher.finish()
     }
 }
+
+#[cfg(test)]
+impl PartialEq for HashTable {
+    // Checks if `HashTable` are equivalent
+    fn eq(&self, other: &Self) -> bool {
```

Comment:
This implementation feels pretty awkward. Is the manual implementation to compare self.data and other.data necessary?

Reply:
Before when I had `self.data == other.data` one of the tests was failing and I suspected this was because I was compared Boxed structures. However, now I can't recreate the problem, so I will change it back to `self.data == other.data`

Reply:
I have now seen the test `new_file_backed_cache_changed_and_restored` fail again at the `assert!(new_cache == old_cache);` and I added extra print statements to see which part failed, except I cannot reproduce this fail.